pkg/planner/core/find_best_task.go

```diff
@@ -2597,6 +2597,10 @@ func convertToTableScan(ds *logicalop.DataSource, prop *property.PhysicalPropert
 	var task base.Task = copTask
 	if candidate.isMatchProp {
 		copTask.keepOrder = true
+		if ds.Table.Type().IsClusterTable() {
```

Comment:
we can directly to modify the variable of isMatchProp in
![image](https://github.com/user-attachments/assets/a5186553-a0fc-4150-9163-ad436a4583b8)

Reply:
Done